### PR TITLE
Attempt 3 to setup Lerna

### PR DESCRIPTION
**Pull request by macdiesel**
**Merge checklist:**
- [ ] Evaluate how your changes will impact existing consumers (e.g., `frontend-app-learner-portal-enterprise`, `frontend-app-admin-portal`, and `frontend-app-enterprise-public-catalog`). Will consumers safely be able to upgrade to this change without any breaking changes?
- [ ] Ensure your commit message follows the semantic-release conventional commit message format. If your changes include a breaking change, ensure your commit message is explicitly marked as a `BREAKING CHANGE` so the NPM package is released as such.
- [ ] Once CI is passing, verify the package versions that Lerna will increment to in the Github Action CI workflow logs.
    - *Note*: This may be found in the "Preview Updated Versions (dry run)" step in the Github Action CI workflow logs.

**Post merge:**
- [ ] Verify Lerna created a release commit (e.g., ``chore(release): publish``) that incremented versions in relevant package.json and CHANGELOG files, and created [Git tags](https://github.com/openedx/frontend-enterprise/tags) for those versions.
- [ ] Run the ``Publish from package.json`` Github Action [workflow](https://github.com/openedx/frontend-enterprise/actions/workflows/publish-from-package.yml) to publish these new package versions to NPM.
    - This may be triggered by clicking the "Run workflow" option for the ``master`` branch.
- [ ] Verify the new package versions were published to NPM (i.e., ``npm view <package_name> versions --json``).
    - *Note*: There may be a slight delay between when the workflow finished and when NPM reports the package version as being published. If it doesn't appear right away in the above command, try again in a few minutes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ jobs:
     name: Build
     runs-on: ubuntu-latest
     steps:
+    - name: GitHub Action for npx
+      uses: mikeal/npx@1.0.0
```
Comment (adamstankiewicz) [inline .github/workflows/ci.yml:12]: Is this step necessary? `npx` should be able to run within Github Actions without this. We run `npx` in other Github Action workflow files within other repos without this step.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,14 +38,11 @@ jobs:
...
+    - name: "[Lerna] Preview Updated Versions (dry run)"
       # switch from detatched head to a real branch so we can pass it to `--allow-branch`
       run: |
         git switch -c "actual-branch-not-detached-head"
-        echo "Installing commitlint"
-        npm install --save commitlint
-        npm install --save conventional-changelog-conventionalcommits
-        echo "Finished installing Installing commitlint"
-        lerna version --allow-branch actual-branch-not-detached-head --no-git-tag-version --no-changelog --no-push --yes
+        npx lerna init
```
Comment (adamstankiewicz) [inline .github/workflows/ci.yml:45]: I believe `lerna init` is typically only run to generate the `lerna.json` configuration file. Out of curiosity, does this work without `lerna init`? If not, are there changes generated from this `lerna init` command that should be actually persisted/committed in the `lerna.json` file instead?

File: .github/workflows/publish-from-package.yml [hunk 3]
```diff
@@ -38,6 +40,8 @@ jobs:
       - name: Coverage Report
         uses: codecov/codecov-action@v3
       - name: Publish to NPM from package.json versions
-        run: lerna publish from-package --yes
+        run: |
+          npx lerna init
```
Comment (adamstankiewicz) [inline .github/workflows/publish-from-package.yml:44]: Same question about `lerna init` here.

File: .github/workflows/publish-from-package.yml [hunk 4]
```diff
@@ -8,6 +8,8 @@ jobs:
     if: "!(github.actor == 'github-actions[bot]') "
     runs-on: ubuntu-latest
     steps:
+      - name: GitHub Action for npx
+        uses: mikeal/npx@1.0.0
```
Comment (adamstankiewicz) [inline .github/workflows/publish-from-package.yml:12]: Same question about whether this Github Action is necessary.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -40,6 +42,8 @@ jobs:
         run: npm run test
       - name: Coverage Report
         uses: codecov/codecov-action@v3
+      - name: Install lerna via npx
+        run: npx lerna init
```
Comment (adamstankiewicz) [inline .github/workflows/release.yml:46]: Same question about `lerna init`.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -11,6 +11,8 @@ jobs:
     name: Release
     runs-on: ubuntu-latest
     steps:
+      - name: GitHub Action for npx
+        uses: mikeal/npx@1.0.0
```
Comment (adamstankiewicz) [inline .github/workflows/release.yml:15]: Same question about whether this step is necessary.

File: package.json [hunk 7]
```diff
@@ -22,9 +22,10 @@
     "bootstrap": "npm install conventional-changelog-conventionalcommits",
     "changed": "lerna changed"
   },
-  "devDependencies": {
-    "@commitlint/config-conventional": "13.2.0",
+"devDependencies": {
```
Comment (adamstankiewicz) [inline package.json:25]: Lost the indent.

File: package.json [hunk 8]
```diff
@@ -22,9 +22,10 @@
...
+    "@commitlint/config-conventional": "17.6.0",
+    "conventional-changelog-conventionalcommits": "^5.0.0",
     "husky": "6.0.0",
-    "lerna": "4.0.0"
+    "lerna": "6.6.1"
```
Comment (adamstankiewicz) [inline package.json:29]: [clarification] Did you look through the changelog to see what the specific breaking changes are for v5 and v6? Wondering if there are any breaking changes we need to account for?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (macdiesel) [reply .github/workflows/ci.yml]: This was me not fully understanding npx, npm and lerna. I've read the lerna docs and understand better now what this does. Updated in latest PR.

File: .github/workflows/publish-from-package.yml (hunk 3 above)
Comment (macdiesel) [reply .github/workflows/publish-from-package.yml]: Removed

File: .github/workflows/release.yml (hunk 5 above)
Comment (macdiesel) [reply .github/workflows/release.yml]: Removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (macdiesel) [reply .github/workflows/ci.yml]: Not necessary, removed.

**Review by adamstankiewicz — APPROVED** [review]
LGTM. My only remaining feedback is whether we should pin the `lerna` command when running it with `npx` (e.g., `npx lerna@6`). That way, when a new major version of `lerna` is released, CI will still use the same version that is currently installed in the package.json and de-risks future breaking changes.

**Pull request merged**